app/bundles/CampaignBundle/Tests/Command/TriggerCampaignCommandTest.php

```diff
@@ -535,8 +535,36 @@ public function testCampaignActionChangeMembership(): void
         $campaign2 = $this->createCampaign('Campaign 2');
         $lead      = $this->createLead('Lead');
         $this->createCampaignLead($campaign1, $lead);
+        $this->em->flush();
+        $property = ['addTo' => [$campaign2->getId()], 'removeFrom' => ['this']];
+        $this->createEvent('Event', $campaign1, 'campaign.addremovelead', 'action', $property);
+        $this->em->flush();
+        $this->em->clear();
+
+        $this->testSymfonyCommand('mautic:campaigns:trigger', ['--campaign-id' => $campaign1->getId(), '--contact-id' => $lead->getId(), '--kickoff-only' => true]);
+
+        $campaignLeads = $this->em->getRepository(Lead::class)->findBy(['lead' => $lead], ['campaign' => 'ASC']);
+
+        Assert::assertCount(2, $campaignLeads);
+        Assert::assertSame($campaign1->getId(), $campaignLeads[0]->getCampaign()->getId());
+        Assert::assertTrue($campaignLeads[0]->getManuallyRemoved());
+        Assert::assertSame($campaign2->getId(), $campaignLeads[1]->getCampaign()->getId());
+        Assert::assertFalse($campaignLeads[1]->getManuallyRemoved());
+    }
+
+    public function testCampaignActionChangeMembershipRestartRotation(): void
+    {
+        $campaign1 = $this->createCampaign('Campaign 1');
+        // create campaign with restart allowed
+        $campaign2 = $this->createCampaign('Campaign 2', true);
```

Comment:
You can change the campaign in place, allowing later to grasp what's going on without a need to review method signature.

```suggestion
        $campaign2 = $this->createCampaign('Campaign 2');
        $campaign2->setAllowRestart(true);
```